android/modules/ui/src/java/ti/modules/titanium/ui/widget/tabgroup/TiUIAbstractTabGroup.java

```diff
@@ -135,6 +135,9 @@
 	protected boolean tabsDisabled = false;
 	protected int numTabsWhenDisabled;
 	protected int colorPrimaryInt;
+	// Max number of tabs supported. Currently in use for BottomNavigation style only.
+	// Default value of -1 is used to represent no tab limit.
+	protected int maxTabsSupported = -1;
```

Comment:
Instead of a member variable, how about we have a `getMaxTabsSupported()` method instead. By default, it can return `Intent.MAX_VALUE` like so. Setting it to max int means we can easily do greater-than checks.
```java
public int getMaxTabsSupported()
{
	return Integer.MAX_VALUE;
}
```

Then "TiUIBottomNavigationTabGroup" can then override it like this...
```java
@Override
public int getMaxTabsSupported()
{
	return 5;
}
```

Ideally, we should call [BottomNavigationView.getMaxItemCount()](https://developer.android.com/reference/android/support/design/widget/BottomNavigationView#getmaxitemcount) to get the maximum, but unfortunately it is an instance method, not a static method.

Reply:
Hold on. Maybe we don't need a `maxTabsSupported` member variable or `getMaxTabsSupported()` method.

A simpler solution might be to have `TiUIBottomNavigationTabGroup` override the `addTab()` method like this?
```java
@Override
public void addTab(TabProxy tabProxy)
{
	if (this.mBottomNavigationView == null) {
		return;
	}

	final int MAX_TABS = this.mBottomNavigationView.getMaxItemCount();
	if (this.tabs.size() < MAX_TABS) {
		super.addTab(tabProxy);
	} else {
		Log.w(TAG, "Bottom style TabGroup cannot have more than " + MAX_TABS + " tabs.");
	}
}
```

Reply:
I have totally missed the `getMaxItemCount()` method for the bottom navigation view. Moved to using it and override the `addTab()` method. It is a far better and clean solution. Thanks!